### DISCUSSION TITLE
How to display the screen for asking user's permission to use his/her personal data in Vue.js mobile app in App Store?

### DISCUSSION THREAD

**Discussion by tesicg** [quasarframework/quasar#9636] [General - Components / Directives / etc]
Beginnning from iOS 14.5, Apple needs from mobile apps in App Store to display some screen that should ask for user's permission to use his/her personal data.

It seems that it's in connection with this article:

https://developer.apple.com/documentation/bundleresources/information_property_list/nsusertrackingusagedescription/

I've found some code written in Swift, but it doesn't help me too much.

I can't find on Internet anything that explains how to do it in mobile app written in Vue.js (Quasar framework).

I believe there's someone who can help with this issue.

**Comment by ontwikkelfabriek** [comment]
@tesicg 

> Beginnning from iOS 14.5, Apple needs from mobile apps in App Store to display some screen that should ask for user's permission to use his/her personal data.

You only have to do that **IF** you are tracking the person's or the person's device. So do you do that?

If you do ( or need other stuff like the camera) than you have to edit the config.xml( cordova) in `src-cordova/config.xml'. Here you can add a plist key( for the required permission) and add (your own app specific) description why your app needs this. 

example from my cordova app ( config.xml):
```xml
...
<platform name="ios">
        <config-file parent="NSLocationAlwaysUsageDescription" target="*-Info.plist">
            <string>need location access to find bee locations nearby on map </string>
        </config-file>
        <config-file parent="NSLocationWhenInUseUsageDescription" target="*-Info.plist">
            <string>need location access to find bee locations nearby on map</string>
        </config-file>
        <config-file parent="NSPhotoLibraryUsageDescription" target="*-Info.plist">
            <string> need Photo Libary access to select images of bee friendly areas</string>
        </config-file>
        <config-file parent="NSCameraUsageDescription" target="*-Info.plist">
            <string>Need Camera access to take images of bee friendly areas</string>
        </config-file>
         ...
```

**Reply by tesicg** [reply]
"You only have to do that IF you are tracking the person's or the person's device. So do you do that?"

Yes, we do that.

Does it mean if we want to display that dialog we need to add the following entry:

```
<config-file target="*-Info.plist" parent="NSUserTrackingUsageDescription">
    <string>The application requires permission to process your personal data, without which it cannot work</string>
</config-file>
```

**Reply by ontwikkelfabriek** [reply]
@tesicg 
>Does it mean if we want to display that dialog we need to add the following entry:

```
<config-file target="*-Info.plist" parent="NSUserTrackingUsageDescription">
    <string>The application requires permission to process your personal data, without which it cannot work</string>
</config-file>
```

Yes, but the description must **not** be **generic** but specific for your app( the desciption in your sample is **VERY** generic and will not be accepted). The description has to explain why **your** app needs those permission. (See my sample descriptions , they are specific to my app.)

You need those entries mostly to pass the app review. Some of the plist descriptions, like `NSUserTrackingUsageDescription`, will also be used to show an alert request by Apple/Ios to inform the user why the app is requesting permission to use data for tracking the user or the device.  The dialog that displays the `system-permission alert request` will be managed by apple/ios, you don't have to code that. You just have to make sure your create the necessary plist properties and the descriptions that match your app's 'needs'.

**Comment by tesicg** [comment]
"The description has to explain why your app needs those permission."

Do you mean mentioning the word "bee" in your case?

Do I need in my case use English or Russian language?

**Reply by ontwikkelfabriek** [reply]
@tesicg 
> Do you mean mentioning the word "bee" in your case?

Ha yes indeed , the app is about bees, people can mark locations on a map that have a good environment for bees. 

> Do I need in my case use English or Russian language?

That's a good question. My app was only published 1 language(dutch) and I used english for all of my permissions/descriptions, also my descriptions where not displayed in a dialog to the user but just to pass the app review.

Here's how you can do multi language plist in cordova:
https://www.npmjs.com/package/cordova-plugin-localization-strings

**Reply by tesicg** [reply]
I've tried to change the description several times, but it still doesn't work.

We've found this in the meantime:

https://github.com/chemerisuk/cordova-plugin-idfa

But, I'm not sure how to use it in our Vue.js (Quasar) app. There is some example, but I don't understand where should I put that code.

**Reply by ontwikkelfabriek** [reply]
@tesicg 

>  but it still doesn't work.

what does not work specifically? 

> But, I'm not sure how to use it in our Vue.js (Quasar) app. There is some example, but I don't understand where should I put that code.

I would create another topic about that.

**Reply by tesicg** [reply]
> @tesicg
> 
> > But, I'm not sure how to use it in our Vue.js (Quasar) app. There is some example, but I don't understand where should I put that code.
> 
> I would create another topic about that.

Ok. I'll do it.

**Reply by tesicg** [reply]
> @tesicg
> 
> > but it still doesn't work.
> 
> what does not work specifically?

Here's what we got from App Store:

https://drive.google.com/file/d/155BAXAOPg3ccZE1SoneAW7FdP1GLbT7w/view?usp=sharing

https://drive.google.com/file/d/1kbHIFSy0HOlyGClObETRNKxmGlaOGlAO/view?usp=sharing

**Reply by ontwikkelfabriek** [reply]
@tesicg 

>Here's what we got from App Store:

Ah I see, Apple created yet another reason to not pass the app review...

Yes I guess you have to implement the `idfa` cordova plugin in your app ( see your post)
